src/Middleware/Session/src/DistributedSession.cs

```diff
@@ -289,7 +289,9 @@ public async Task CommitAsync(CancellationToken cancellationToken = default)
                 {
                     try
                     {
+                       if(!_isNewSessionKey){
```

Comment:
You can make the else on line 288 into an else if with this condition.

Reply:
Okay, I'll take care of it later